src/Microsoft.AspNet.Hosting/Server/ServerLoader.cs

```diff
@@ -18,59 +18,29 @@ public ServerLoader(IServiceProvider services)
             _services = services;
         }
 
-        public IServerFactory LoadServerFactory(string serverFactoryIdentifier)
+        public IServerFactory LoadServerFactory([NotNull] string assemblyName)
         {
-            if (string.IsNullOrEmpty(serverFactoryIdentifier))
+            if (string.IsNullOrEmpty(assemblyName))
             {
-                throw new ArgumentException(string.Empty, nameof(serverFactoryIdentifier));
+                throw new ArgumentException(string.Empty, nameof(assemblyName));
             }
 
-            var nameParts = HostingUtilities.SplitTypeName(serverFactoryIdentifier);
-            var typeName = nameParts.Item1;
-            var assemblyName = nameParts.Item2;
-
             var assembly = Assembly.Load(new AssemblyName(assemblyName));
             if (assembly == null)
             {
                 throw new Exception(string.Format("The assembly {0} failed to load.", assemblyName));
             }
 
-            Type type = null;
-            Type interfaceInfo;
-            if (string.IsNullOrEmpty(typeName))
-            {
-                foreach (var typeInfo in assembly.DefinedTypes)
-                {
-                    interfaceInfo = typeInfo.ImplementedInterfaces.FirstOrDefault(interf => interf.Equals(typeof(IServerFactory)));
-                    if (interfaceInfo != null)
-                    {
-                        type = typeInfo.AsType();
-                    }
-                }
+            var serverType = assembly.DefinedTypes.Where(
+                t => t.ImplementedInterfaces.FirstOrDefault(interf => interf.Equals(typeof(IServerFactory))) != null)
```

Comment:
Ah thanks, I don't think we use this entry point very often, I'm not sure we have any unit test coverage at all currently of this...  I'll see if I can find someplace to add one for this